### ISSUE TITLE
Crypto.Utils.strxor_c takes (bytes, int) instead of (bytes, bytes)

### ISSUE THREAD

**Issue by aytimothy** [Legrandin/pycryptodome#618]
As per the latest version, try running this code:

```python
from Crypto.Util import strxor
encrypted_bytes = bytes([1, 2, 3, 4, 252, 253, 254, 255])
cypher_asrepeatedlist = bytes([255, 255, 255, 255, 255, 255, 255, 255])
cypher_asbytearray = bytes([255])
print("Test 1: " + str(strxor.strxor(encrypted_bytes, cypher_asrepeatedlist)))
print("Test 2: " + str(strxor.strxor_c(encrypted_bytes, 255)))
print("Test 3: " + str(strxor.strxor_c(encrypted_bytes, cypher_asbytearray)))
```

Expected output:
```
Test 1: b'\xfe\xfd\xfc\xfb\x03\x02\x01\x00'
Test 2: b'\xfe\xfd\xfc\xfb\x03\x02\x01\x00'
Test 3: b'\xfe\xfd\xfc\xfb\x03\x02\x01\x00'
```

Got output: 
```
Test 1: b'\xfe\xfd\xfc\xfb\x03\x02\x01\x00'
Test 2: b'\xfe\xfd\xfc\xfb\x03\x02\x01\x00'
Traceback (most recent call last):
  File "D:\Users\aytimothy\PycharmProjects\digirearise\xorfail.py", line 7, in <module>
    print("Test 3: " + str(strxor.strxor_c(encrypted_bytes, cypher_asbytearray)))
  File "C:\Program Files\Python39\lib\site-packages\Crypto\Util\strxor.py", line 106, in strxor_c
    if not 0 <= c < 256:
TypeError: '<=' not supported between instances of 'int' and 'bytes'
```

I'm using PyCryptoDome version 3.14.1 and Python version 3.9.6.

---

Also, just to make sure I'm reading it right, when I use `strxor_c` on (assume these examples are byte arrays) with the same "encrypted bytes" as the example above...

* [255] as the key is the same as regular `strxor`ing [255, 255, 255, 255, 255, 255, 255, 255, 255] and
* [0, 63, 127, 191] as the key is the same as `strxor`ing [0, 63, 127, 191, 0, 63, 127, 191]

**Comment by Legrandin** [comment]
Fixed with 46746e7bb50d3148d15ae2f084e6a47a67af1f44

**Comment by Legrandin** [comment]
Fixed in v3.15.0
